cmd/oras/internal/errors/errors.go

```diff
@@ -60,6 +65,45 @@ func CheckArgs(checker func(args []string) (bool, string), Usage string) cobra.P
 	}
 }
 
+// Handler handles error during cmd execution.
+type Handler interface {
+	// Handle handles error during cmd execution.
+	// if nil is returned, then err cannot be handled.
+	Handle(err error, cmd *cobra.Command) error
```

Comment:
nit: naming convention
```suggestion
	Handle(cmd *cobra.Command, err error) error
```

Reply:
Example: https://pkg.go.dev/io/fs#WalkDirFunc
```go
type WalkDirFunc func(path string, d DirEntry, err error) error
```